src/fpm_command_line.f90

```diff
@@ -107,8 +109,16 @@ module fpm_command_line
 !> Settings for interacting and updating with project dependencies
 type, extends(fpm_cmd_settings)  :: fpm_update_settings
     character(len=ibug),allocatable :: name(:)
-    logical :: fetch_only
-    logical :: clean
+    character(len=:),allocatable    :: dump
+    logical                         :: fetch_only
+    logical                         :: clean
+end type
+
+!> Settings for exporting model data
+type, extends(fpm_build_settings) :: fpm_export_settings
+    character(len=:),allocatable  :: dump_manifest
+    character(len=:),allocatable  :: dump_dependencies
+    character(len=:),allocatable  :: dump_model
```

Comment:
Why not using `logical`s?

Reply:
Because the user can specify an optional file name

Reply:
I think it's cleaner to add another field called `output_file` and use `logical`s here (separation of concerns).

Instead of:

```fortran
if (len_trim(settings%dump)>0) then
  call deps%dump(trim(settings%dump), error, json=name_is_json(trim(settings%dump)))
  call handle_error(error)
end if
```

You'd have something like:

```fortran
if (settings%dump) then
  call deps%dump(settings%output_file, error)
  call handle_error(error)
end if
```

Reply:
I’m not sure I agree, this is a CLI structure and we want to keep it simple. However, I like the idea of putting in a more clean structure! Can I ask you to take care of that @minhqdao?